### ISSUE TITLE
should discover sampling time in ODF file for CTD

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1142]
I am not sure whether the file source is public, so I won't state it here. CR knows what I'm talking about, though. I changed the filename to `a.odf` and I see that `read.oce()` is not discovering the sampling time. There are a *lot* of items in the header that would seem to indicate sampling time, but some are clearly wrong. I'd like `oce` to be able to determine the date automatically, without asking the user to make a decision. 

Below is what I get from looking at ODF header comments (which use upper case) and then at the Seabird header comments.  From this information I am tempted to guess that `CREATION_DATE` is the thing that `oce` should take as the date. Any thoughts, CR?

(PS. the source for ODF format is "Anthony W. Isenor and David Kellow, 2011. ODF Format Specification Version 2.0" but I can no longer find that online. I have a local copy, and when I looked for `ORIG_CREATION_DATE` in that .doc file, I found it just twice -- in neither case with any embedding text to tell what this means, as opposed to the other time entries.)

ODF:
```
grep DATE a.odf
   START_DATE='01-Jan-2016 00:00:00.00',
   END_DATE='31-Dec-2016 23:59:59.00',
   CREATION_DATE='14-DEC-2016 12:26:22.00',
   ORIG_CREATION_DATE='14-DEC-2016 12:26:22.00',
   START_DATE_TIME='14-DEC-2016 13:19:05.00',
   END_DATE_TIME='17-NOV-1858 00:00:00.00',
   CREATION_DATE='14-DEC-2016 12:26:22.00',
   CREATION_DATE='14-DEC-2016 12:26:22.00',
```

Seabird:
```
grep Time a.odf
   PROCESS='* System UpLoad Time = Dec 14 2016 12:16:54',
```

**Comment by richardsc** [comment]
Sorry for being slow to reply to this. I think it might require a closer look at the file, including the original raw file, because I can't tell which one is actually supposed to be the time associated with the CTD cast.

Generally the "System UpLoad Time" is the time that the data was transferred from the CTD to a computer, and isn't related at all to the profile, so I wouldn't plan to use that one. Also, it looks to me like there might be some time zone confusion between those times, because I don't think it's likely that the "System Upload time" was from earlier than the actual profile time (e.g. maybe the `CREATION_DATE`). Probably one is local (System upload?) and the other is UTC.

I'll look into this some more, at least for this example, and hopefully that will tell us more about how to read the times generally.

**Comment by dankelley** [comment]
It's looking to me as though the right thing to check is `START_DATE_TIME`. At least that seems reasonable based on the following tests (with files now pushed to GH).

```
cd tests/testthat/local_data/bedford_basin
git grep -i start_time *
bedford_basin[develop *]$ git grep -i start_time *
CTD_BCD2010667_001_01_DN.ODF:  PROCESS='# start_time = Jan 05 2010 13:23:29',
CTD_BCD2011667_001_01_DN.ODF:  PROCESS='# start_time = Jan 05 2011 09:10:36',
CTD_BCD2012667_01_01_DN.ODF:  PROCESS='# start_time = Jan 04 2012 12:53:38 [Instrument's time stamp, header]',
CTD_BCD2013667_001_01_DN.ODF:  PROCESS='# start_time = Jan 02 2013 15:04:39 [Instrument's time stamp, header]',
CTD_BCD2014667_001_01_DN.ODF:  PROCESS='# start_time = Jan 08 2014 13:37:15 [Instrument's time stamp, header]',
CTD_BCD2015667_001_01_DN.ODF:  PROCESS='# start_time = Jan 07 2015 13:28:34 [Instrument's time stamp, header]',
D16667001.ODF:   PROCESS='# start_time = Jan 06 2016 13:17:37 [Instrument's time stamp, header]',
bedford_basin[develop *]$ 
bedford_basin[develop *]$ 
bedford_basin[develop *]$ git grep -i start_date_time *
CTD_BCD2010667_001_01_DN.ODF:  START_DATE_TIME='05-JAN-2010 13:23:29.00',
CTD_BCD2011667_001_01_DN.ODF:  START_DATE_TIME='05-JAN-2011 09:10:36.00',
CTD_BCD2012667_01_01_DN.ODF:  START_DATE_TIME='04-JAN-2012 12:53:38.00',
CTD_BCD2013667_001_01_DN.ODF:  START_DATE_TIME='02-JAN-2013 15:04:39.00',
CTD_BCD2014667_001_01_DN.ODF:  START_DATE_TIME='08-JAN-2014 13:37:15.00',
CTD_BCD2015667_001_01_DN.ODF:  START_DATE_TIME='07-JAN-2015 13:28:34.00',
D16667001.ODF:   START_DATE_TIME='06-JAN-2016 13:17:37.00',
```

Therefore, I will code it so that `START_DATE_TIME` becomes `metadata$startTime`, which is how it has been named in other `ctd` objects. (I can always change this later, if need be.)

**Comment by dankelley** [comment]
Fixed in `develop` branch, commit f578b576a2aedc1f0a27d27667cb9baec720a6ef

**Comment by richardsc** [comment]
If I had to guess, I think I'd agree with you. But that guess would be assuming that the `START_DATE_TIME` is in UTC, and the `System Upload Time` is local ... ugh, time zones ...
